src/ignore_gradients.jl

```diff
@@ -0,0 +1,50 @@
+"""
+    ignore_gradients(f::Function)
+
+Tells the AD system to ignore the gradients of the wrapped closure. The primal computation
+(forward pass) is executed normally.
+
+```julia
+ignore_gradients() do
+    value = rand()
+    push!(collection, value)
+end
+```
+
+Using this incorrectly could lead to incorrect gradients. E.g.
+```julia
+function wrong_grads(x)
+    y = ones(3)
+    ignore_gradients() do
+        push!(y, x)
+    end
+    return sum(y)
+end
+```
+"""
+ignore_gradients(f::Function) = f()
+
+"""
+    ignore_gradients(x)
+
+Tells the AD system to ignore the gradients of the argument. Can be used to avoid
+unnecessary computation of gradients.
+
+```julia
+ignore_gradient(x) * w
+```
+"""
+ignore_gradients(x) = x
+
+@non_differentiable ignore_gradients(f)
+
+"""
+    @ignore_gradients (...)
+
+Tell the AD system to ignore the expression. Equivalent to `ignore_gradients() do (...) end`.
```

Comment:
```suggestion
Tells the AD system to ignore the expression. Equivalent to `ignore_gradients() do (...) end`.
```
for consistency with previous docstring

Reply:
Should be the other way around actually, see https://github.com/JuliaDiff/ChainRulesCore.jl/pull/229/files#r716405173

Reply:
I actually prefer it this way, since it describes what the function does, rather than what you can do with the function.